x/exchange/keeper/orders.go

```diff
@@ -812,3 +812,25 @@ func (k Keeper) IterateAddressOrders(ctx sdk.Context, addr sdk.AccAddress, cb fu
 func (k Keeper) IterateAssetOrders(ctx sdk.Context, assetDenom string, cb func(orderID uint64, orderTypeByte byte) bool) {
 	k.iterateOrderIndex(ctx, GetIndexKeyPrefixAssetToOrder(assetDenom), cb)
 }
+
+// CancelAllOrdersForMarket cancels all orders for a market, deleting them and releasing their holds.
+func (k Keeper) CancelAllOrdersForMarket(ctx sdk.Context, marketID uint32, signer string) {
+	var orderIDs []uint64
+	k.IterateMarketOrders(ctx, marketID, func(orderID uint64, _ byte) bool {
+		orderIDs = append(orderIDs, orderID)
+		return false
+	})
+
+	var errs []error
+	for _, orderID := range orderIDs {
```

Comment:
Can you just move this into the body of the iterate function? Seems kind of odd to iterate twice if we aren't extracting out the first part for readability.

Reply:
Nope. The cancellation of an order deletes that entry from state and its related index entries. it's bad practice to alter the thing you're iterating over as you're iterating over it.

Reply:
Ah, it wasn't immediately obvious that it was modifying the iterator from the name. It looked like it was just modifying the content of the iterated elements.